js/tiles.js

```diff
@@ -379,6 +379,54 @@ const door = {
       this.ground.show();
     },
   }),
+  river: Object.assign({}, OpenDoors, {
+      canEnterFromTheRight() {return false;},
+      canLeaveToTheRight() {return false;},
+      canEnterFromTheLeft() {return false;},
+      canLeaveToTheLeft() {return false;},
```

Comment:
If the player cannot enter, how can the player visit the tile?
I think, the first argument to this function is always "player" although it is not described here. JavaScript allows not using arguments. 

You could do something like: 
```
canEnterFromTheRight() {
return player.inventory.has('Boat')
} 

```